cmd/soroban-cli/src/events.rs

```diff
@@ -0,0 +1,191 @@
+use std::io;
+use std::io::Write;
+
+use clap::Parser;
+use hex::FromHexError;
+use soroban_env_host::xdr::{ReadXdr, ScVal};
+use termcolor::{Color, ColorChoice, ColorSpec, StandardStream, WriteColor};
+
+use crate::rpc::Event;
+use crate::utils;
+use crate::{rpc, rpc::Client};
+use crate::{HEADING_RPC, HEADING_SANDBOX};
+
+#[derive(Parser, Debug)]
+#[clap()]
+pub struct Cmd {
+    /// The ledger range to pull events from
+    start_ledger: u32,
+    end_ledger: u32,
+
+    /// RPC server endpoint
+    #[clap(long,
+        env = "SOROBAN_RPC_URL",
+        help_heading = HEADING_RPC,
+        conflicts_with = "sandbox",
+    )]
+    rpc_url: Option<String>,
+
+    /// Sandbox file from which to pull events from
+    #[clap(
+        long,
+        parse(from_os_str),
+        value_name = "PATH",
+        env = "SOROBAN_LEDGER_FILE",
+        help_heading = HEADING_SANDBOX,
+        conflicts_with = "rpc-url",
+    )]
+    ledger_file: Option<std::path::PathBuf>,
+
+    /// A set of (up to 5) contract IDs to filter events on
+    #[clap(long = "ids", multiple = true)]
```

Comment:
❗ 
```suggestion
    #[clap(long = "id", multiple = true)]
```
In other subcommands we've named multi-value options without the plural, since that on the command line they'll be expressed one at a time. For example, if it is named `ids`, it will look like this:
```
soroban events --ids 1 --ids 2 --ids 3 ...
```

For consistency with existing multi-value options I think we should rename it to `id`:

```
soroban events --id 1 --id 2 --id 3
```

Reply:
Oh nice. I was doing it as `--ids 1 2 3` which is why the plural made sense; I didn't realize you can repeat the argument name like that.